Core.DomainServices/Organizations/OrgUnitService.cs

```diff
@@ -87,8 +95,17 @@ public bool IsAncestorOf(int unitId, int ancestorId)
             return IsAncestorOf(unit, ancestor);
         }
 
-        public void Delete(int id)
+        public void Delete(int id, int organizationId)
         {
+            using var transaction = _transactionManager.Begin();
+
+            var organization = _organizationRepository.GetByKey(organizationId);
+            var result = organization.RemoveOrganizationUnit(id);
```

Comment:
this implementation is not using the knowledge of the new "registrations" service...
Create a new domain command, the handler should then call the "getregistrations" and then "delete registrations"

After that removal of the unit can proceed

Reply:
a lot of the existing code here should die since it must now be handled in the command handler